[paymentservice] add basic metrics support

pichlermarc opened the pull request:
Related to #68

## Changes

This PR adds a counter `app.payment.transactions` that can be split by the transaction currency, adds dependencies to the JS Metrics packages, and changes the startup process so that the async resource detection can finish before creating a `Meter` or `Tracer`.

Once updated instrumentation packages are released, I will open a follow-up PR that updates the metrics API/SDK to the GA versions. Then the added temporality environment variable will also change the temporality of the exported metrics accordingly. :slightly_smiling_face: 

![image](https://user-images.githubusercontent.com/22105064/201944566-d41e32e0-ad8b-4687-9716-dfd85d263f8f.png)

julianocosta89 reviewed (approved):
Nice one!
![image](https://user-images.githubusercontent.com/15364991/201952387-d49a9190-093e-43fe-a7ee-349674f36134.png)

cartersocha commented:
@pichlermarc would you mind updating our metric feature matrix too? Lgtm otherwise. Thanks for this!

https://github.com/open-telemetry/opentelemetry-demo/blob/main/docs/metric_service_features.md

cartersocha commented:
Oh and we'll need to update the payment service doc in the docs/services folder

pichlermarc commented:
@cartersocha changed some text to the payment service docs to reflect the new setup, added some text for metrics, and changed the entry in the matrix :slightly_smiling_face:

mviitane commented on docs/services/paymentservice.md:
```diff
@@ -8,33 +8,39 @@ processed.
-code required to initialize the SDK and auto-instrumentation based on standard
-OpenTelemetry environment variables for OTLP export, resource attributes, and
-service name.
+It is recommended to use wrap your app using an OpenTelemetry initialization
```
The wrap concept could be explained a bit more. For example, for me, it's not clear if we use wrapping in the code below.

pichlermarc replied:
Tried to clarify the approach in [b626a9c](https://github.com/open-telemetry/opentelemetry-demo/pull/583/commits/b626a9c80d23b13a8b99be475e9093e4aa0c749b). :slightly_smiling_face:

cartersocha commented:
Thanks again @pichlermarc! Merging now
